### PR TITLE
[Android] Split adding projects and account manager to different screens

### LINKED ISSUE #4182: [Android] Refactor 'Add Account Manager' action
* 'Add Account Manager' should be next to 'Add Project' item in main menu
* Remove 'Add Account Manager' from 'Add Project' screen
* Add new screen with two buttons: 
* * 'Add Project'
* * 'Add Account Manager'
* This new screen should be shown when no Projects (or Account Manager) are added

### PR DESCRIPTION
**Pull request by AenBleidd**
This fixes #4182

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

**Comment by AenBleidd** [comment]
Screenshot_2021-04-18-23-38-30-966_edu berkeley boinc
Screenshot_2021-04-18-23-38-37-041_edu berkeley boinc
Screenshot_2021-04-18-23-38-43-225_edu berkeley boinc

**Comment by davidpanderson** [comment]
It should use the same terminology as the desktop GUI, e.g.
"Use account manager"

It should not say "BOINC account manager";
that's confusing, since BAM! (BOINC Account Manager) is one account manager but there are others.
It should not say "add and manage multiple projects", since some account managers
(e.g. Science United) don't let you choose specific projects.
The GUI shouldn't show a list of projects with checkboxes;
that's the job of the account manager (via its web interface)

**Comment by AenBleidd** [comment]
@davidpanderson,
> It should use the same terminology as the desktop GUI, e.g.
"Use account manager"
It should not say "BOINC account manager";
that's confusing, since BAM! (BOINC Account Manager) is one account manager but there are others.

Ok, I'll fix that

> It should not say "add and manage multiple projects", since some account managers
(e.g. Science United) don't let you choose specific projects.

I'll fix that also.

> The GUI shouldn't show a list of projects with checkboxes;
that's the job of the account manager (via its web interface)

I think you're a little bit confusing here because the last screenshot shows the screen that user sees when they want to connect to the projects manually without using Account Manager

**Comment by davidpanderson** [comment]
Got it, thanks.

**Comment by AenBleidd** [comment]
Screenshot_2021-04-20-12-44-09-719_edu berkeley boinc
Screenshot_2021-04-20-12-44-16-934_edu berkeley boinc

**Review by talregev — APPROVED** [review]
I test it on armv6 (galaxy gio), and arm (oneplus one). everything working as expected. I approve.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AttachAccountManagerActivity.kt [hunk 1]
```diff
@@ -234,31 +177,71 @@ class AcctMgrFragment : DialogFragment() {
...
+        }
+
+        override fun onServiceDisconnected(className: ComponentName) {
+            // This should not happen
+            attachService = null
+            asIsBound = false
+        }
     }
-}
+}
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AttachAccountManagerActivity.kt]: There is already a new line at the end of file

File: android/BOINC/app/src/main/res/layout/activity_attach.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            android:layout_margin="10dp"
+            android:background="@drawable/shape_button_blue"
+            android:minWidth="150dp"
+            android:onClick="cancelClicked"
+            android:padding="5dp"
+            android:text="@string/confirm_cancel"
+            android:textAppearance="?android:attr/textAppearanceMedium"
+            android:textColor="@android:color/white" />
+
+</RelativeLayout>
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/src/main/res/layout/activity_attach.xml]: There is already a new line at the end of file

**Pull request merged**